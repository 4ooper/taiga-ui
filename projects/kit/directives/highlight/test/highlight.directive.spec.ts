import {Component} from '@angular/core';
import type {ComponentFixture} from '@angular/core/testing';
import {TestBed} from '@angular/core/testing';
import {TuiHighlightDirective} from '@taiga-ui/kit';

describe('TuiHighlight directive', () => {
    let fixture: ComponentFixture<unknown>;

    describe.each([
        {
            name: 'in single occurrence mode',
            isMulti: false,
        },
        {
            name: 'in multi occurrences mode',
            isMulti: true,
        },
    ])('$name', ({isMulti}) => {
        describe.each([
            {
                name: 'with one string variant',
                value: 'ng Ar',
            },
            {
                name: 'with several string variants',
                value: ['Ar', 'Ng'],
            },
            {
                name: 'with one regExp variant',
                value: /[ink]+/gi,
            },
            {
                name: 'with several regExp variants',
                value: [/a/gi, /[ink]+/gi],
            },
        ])('$name', ({value}) => {
            @Component({
                standalone: true,
                imports: [TuiHighlightDirective],
                template: `
                    <div
                        [tuiHighlight]="value"
                        [tuiHighlightMultiOccurrences]="isMulti"
                    >
                        <p>
                            <b>King</b>
                            Arthur
                        </p>
                        <p>King Arthur</p>
                        <p>King Arthur</p>
                        <p>
                            <b>King</b>
                            Arthur
                        </p>
                    </div>
                `,
            })
            class TestComponent {
                public readonly isMulti = isMulti;
                public readonly value = value;
            }

            beforeEach(() => {
                fixture = TestBed.createComponent(TestComponent);
                fixture.detectChanges();
            });

            it('Highlight should shown properly', () => {
                expect(fixture).toMatchSnapshot();
            });
        });
    });
});
