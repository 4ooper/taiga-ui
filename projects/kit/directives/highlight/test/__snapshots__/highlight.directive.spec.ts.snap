// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TuiHighlight directive in multi occurrences mode with one regExp variant Highlight should shown properly 1`] = `
<ng-component
  isMulti={[Function Boolean]}
  value={[Function RegExp]}
>
  <div>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
       Arthur 
    </p>
    <p>
      <mark
        tuihighlightmark=""
      >
        Kin
      </mark>
      g Arthur
    </p>
    <p>
      <mark
        tuihighlightmark=""
      >
        Kin
      </mark>
      g Arthur
    </p>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in multi occurrences mode with one string variant Highlight should shown properly 1`] = `
<ng-component
  isMulti={[Function Boolean]}
  value={[Function String]}
>
  <div>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
    <p>
      Ki
      <mark
        tuihighlightmark=""
      >
        ng Ar
      </mark>
      thur
    </p>
    <p>
      Ki
      <mark
        tuihighlightmark=""
      >
        ng Ar
      </mark>
      thur
    </p>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in multi occurrences mode with several regExp variants Highlight should shown properly 1`] = `
<ng-component
  isMulti={[Function Boolean]}
  value={[Function Array]}
>
  <div>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
      <mark
        tuihighlightmark=""
      >
        A
      </mark>
      rthur 
    </p>
    <p>
      <mark
        tuihighlightmark=""
      >
        Kin
      </mark>
      g 
      <mark
        tuihighlightmark=""
      >
        A
      </mark>
      rthur
    </p>
    <p>
      <mark
        tuihighlightmark=""
      >
        Kin
      </mark>
      g 
      <mark
        tuihighlightmark=""
      >
        A
      </mark>
      rthur
    </p>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
      <mark
        tuihighlightmark=""
      >
        A
      </mark>
      rthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in multi occurrences mode with several string variants Highlight should shown properly 1`] = `
<ng-component
  isMulti={[Function Boolean]}
  value={[Function Array]}
>
  <div>
    <p>
      <b>
        Ki
        <mark
          tuihighlightmark=""
        >
          ng
        </mark>
      </b>
      <mark
        tuihighlightmark=""
      >
        Ar
      </mark>
      thur 
    </p>
    <p>
      Ki
      <mark
        tuihighlightmark=""
      >
        ng
      </mark>
      <mark
        tuihighlightmark=""
      >
        Ar
      </mark>
      thur
    </p>
    <p>
      Ki
      <mark
        tuihighlightmark=""
      >
        ng
      </mark>
      <mark
        tuihighlightmark=""
      >
        Ar
      </mark>
      thur
    </p>
    <p>
      <b>
        Ki
        <mark
          tuihighlightmark=""
        >
          ng
        </mark>
      </b>
      <mark
        tuihighlightmark=""
      >
        Ar
      </mark>
      thur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in single occurrence mode with one regExp variant Highlight should shown properly 1`] = `
<ng-component
  isMulti="false"
  value={[Function RegExp]}
>
  <div>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
       Arthur 
    </p>
    <p>
      King Arthur
    </p>
    <p>
      King Arthur
    </p>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in single occurrence mode with one string variant Highlight should shown properly 1`] = `
<ng-component
  isMulti="false"
  value={[Function String]}
>
  <div>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
    <p>
      Ki
      <mark
        tuihighlightmark=""
      >
        ng Ar
      </mark>
      thur
    </p>
    <p>
      King Arthur
    </p>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in single occurrence mode with several regExp variants Highlight should shown properly 1`] = `
<ng-component
  isMulti="false"
  value={[Function Array]}
>
  <div>
    <p>
      <b>
        <mark
          tuihighlightmark=""
        >
          Kin
        </mark>
        g
      </b>
       Arthur 
    </p>
    <p>
      King Arthur
    </p>
    <p>
      King Arthur
    </p>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;

exports[`TuiHighlight directive in single occurrence mode with several string variants Highlight should shown properly 1`] = `
<ng-component
  isMulti="false"
  value={[Function Array]}
>
  <div>
    <p>
      <b>
        Ki
        <mark
          tuihighlightmark=""
        >
          ng
        </mark>
      </b>
       Arthur 
    </p>
    <p>
      King Arthur
    </p>
    <p>
      King Arthur
    </p>
    <p>
      <b>
        King
      </b>
       Arthur 
    </p>
  </div>
</ng-component>
`;
